chore: release issue template improvements

phisco opened the pull request:
Signed-off-by: Philippe Scorsolini <p.scorsolini@gmail.com>

### Description of your changes

Fixes #3701 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

N.A.

[contribution process]: https://git.io/fj2m9

phisco committed:
chore: release issue template improvements

Signed-off-by: Philippe Scorsolini <p.scorsolini@gmail.com>

jbw976 commented:
Out of curiousity, what does `chore:` mean in the title of this PR? do we have that convention documented anywhere? 🤔

phisco commented:
Yes, @jbw976, sorry I'm used to use that, it's following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/). It's not strongly dictating all the types, so chore is not one of the ones explicitly defined in the spec, but it's a pretty common one. It's usually used to show that it's not touching actual code. Could have been also a docs, probably.

jbw976 commented on .github/ISSUE_TEMPLATE/release.md:
```diff
@@ -18,31 +18,29 @@ Please assign the release manager to the issue.
-this issue for posterity. Refer to this [prior release issue][release-1.7] for
-examples of each step.
+this issue for posterity. Refer to this [prior release issue][release-1.11] for
+examples of each step, assuming release vX.Y.0 is being cut.
```
We should also update the patch release issue template as part of this PR, so both are up to date :)

https://github.com/crossplane/crossplane/blob/master/.github/ISSUE_TEMPLATE/patch_release.md

phisco replied:
I gave it a shot, please have a look at it, I also updated/cleaned up the minor patch release doc once more.

jbw976 commented on .github/ISSUE_TEMPLATE/release.md:
```diff
@@ -18,31 +18,29 @@ Please assign the release manager to the issue.
+- [ ] Published a [new release] for the tagged version, with the same name as the version and descriptive release notes, taking care of generating the changes list selecting as "Previous tag" `vX.Y-1.0`, so the first of the releases for the previous minor.
+- [ ] Updated the [`crossplane/test` repo test workflows][crossplane-test-workflows] to ensure the checkout release branch and helm install version(s) point at the new release
+- [ ] Checked that the [docs release issue] created previously has been completed.
+- [ ] Updated the [releases table] in the `README.md` and the `baseBranches` list in `.github/renovate.json5` on `master`.
```
maybe add back in the extra note we used to have here? e.g. ", removing the now old and unsupported version"

jbw976 replied:
oh and explicitly mention "in a single PR" so folks make sure to streamline here :)

phisco replied:
sounds reasonable, done!

jbw976 reviewed (commented):
looks like this captures most everything in #3701 @phisco, great work!!

Let's clean up a couple things for my comments (esp. doing the patch release issue template also) then get this merged in so we can use it in our next releases 🏁

jbw976 commented on .github/ISSUE_TEMPLATE/release.md:
```diff
@@ -18,46 +18,43 @@ Please assign the release manager to the issue.
-[release-1.7]: https://github.com/crossplane/crossplane/issues/2977
-[Code Freeze]: https://crossplane.io/docs/master/reference/release-cycle.html#code-freeze
\ No newline at end of file
+[xpkg.upbound.io]: https://cloud.upbound.io/browse
```
what do you think about updating this link to https://marketplace.upbound.io/configurations?query=getting-started, both here and in the patch release template?  this could be a more direct way to find the released configurations

jbw976 commented on .github/ISSUE_TEMPLATE/release.md:
```diff
@@ -18,46 +18,43 @@ Please assign the release manager to the issue.
+- [ ] Updated, in a single PR, the following on `master`:
+  - [ ] The [releases table] in the `README.md`, removing the now old unsupported release and adding the new one.
+  - [ ] The `baseBranches` list in `.github/renovate.json5`, removing the now old unsupported release and adding the new one.
+  - [ ] The [`crossplane/test` repo test workflows][crossplane-test-workflows] to ensure the checkout release branch and helm install version(s) point at the new release across all the workflow files.
```
i think this task belongs outside of the parent "in a single PR", because it's in a different repo entirely and therefore must be done in a separate PR 😉

jbw976 commented on .github/ISSUE_TEMPLATE/release.md:
```diff
@@ -18,46 +18,43 @@ Please assign the release manager to the issue.
-- [ ] Updated the [releases table] in the `README.md` on `master`.
-- [ ] Updated the current release version in the [Crossplane docs website repo].
-- [ ] Updated the release branch reaching EOL with [docs removal directive].
-- [ ] Request @jbw976 to remove the EOL docs version from Google Search
```
Actually this step will still be needed I think - I would still go into the google search console and add a removal request for e.g. https://docs.crossplane.io/v1.9

jbw976 reviewed (changes requested):
awesome @phisco thanks for the quick turnaround on the feedback and taking a stab at the patch release template too!

A few more tweaks to make and then we should be good!! 🙇

phisco commented:
@jbw976, addressed your comments, you can have another look at it, let me know if there is anything else, otherwise I'd squash everything before merging it 🙏

jbw976 commented on .github/ISSUE_TEMPLATE/release.md:
```diff
@@ -18,46 +18,44 @@ Please assign the release manager to the issue.
-- [ ] Updated the current release version in the [Crossplane docs website repo].
-- [ ] Updated the release branch reaching EOL with [docs removal directive].
-- [ ] Request @jbw976 to remove the EOL docs version from Google Search
+-- [ ] Request @jbw976 to remove the EOL docs version from Google Search
```
looks like the formatting is broken here, this list item isn't on its own new line - probably because of the double hyphen to start the line?

jbw976 commented on .github/ISSUE_TEMPLATE/patch_release.md:
```diff
@@ -18,31 +18,32 @@ Please assign the release manager to the issue.
+this issue for posterity. Refer to this [prior release issue][release-1.11.1] for
+examples of each step, assuming vX.Y.Z is being cut.
 
 - [ ] Updated all version information in the documentation on the relevant release branch.
```
From https://github.com/crossplane/crossplane/pull/3702#discussion_r1117488438, I think this can be removed too - the docs have nothing at all that needs to be done for a patch release 😅

jbw976 reviewed (approved):
2 final things to address and then please do go ahead and get this squashed and merged! thanks @phisco!!
